Userland/Utilities/pro.cpp

```diff
@@ -182,6 +185,24 @@ ErrorOr<int> serenity_main(Main::Arguments arguments)
             request_headers.set(header.substring_view(0, split.value()), header.substring_view(split.value() + 1));
             return true;
         } });
+    args_parser.add_option(Core::ArgsParser::Option {
+        .argument_mode = Core::ArgsParser::OptionArgumentMode::Required,
+        .help_string = "(HTTP only) Provide basic authentication credentials",
+        .long_name = "auth",
+        .short_name = 'u',
+        .value_name = "username:password",
+        .accept_value = [&](auto* s) {
+            StringView raw_credentials { s, strlen(s) };
+            if (!raw_credentials.contains(':'))
+                return false;
+
+            auto maybe_credentials = String::from_utf8(raw_credentials.trim_whitespace(TrimMode::Both));
```

Comment:
:thinking: what if my password has a space at the end?

Reply:
Darn it. You are completely right. Technically, a username can also be just anything... I'll leave out the trim and add a note.

Reply:
I've adjusted the code accordingly.